### PR TITLE
RDV-368 Transcription request to transcription_api

### PR DESCRIPTION
**Pull request by alexmo16**
The application ask for a transcription of a test file and print the result in the application output console. The transcription of the last video recorded will be done in another task.

### CODE REVIEW

File: steno/src/view/components/top_bar.cpp [hunk 1]
```diff
@@ -92,8 +105,18 @@ void TopBar::onRecorderStateChanged(const QMediaRecorder::State& state)
...
             break;
         case QMediaRecorder::State::StoppedState:
+        {
             m_ui->recordButton->setText("Start recording");
+            const auto transcriptionConfig = m_config->transcriptionConfig();
+            const bool isTranscriptionEnabled =
+                transcriptionConfig->value(Model::TranscriptionConfig::AUTOMATIC_TRANSCRIPTION).toBool();
+            if (isTranscriptionEnabled)
+            {
+                m_transcription->transcribe(QCoreApplication::applicationDirPath() +
```
Comment (Rouiller) [inline steno/src/view/components/top_bar.cpp]: Should we keep this until we can make the call on our recorded files?

File: steno/src/view/components/top_bar.cpp [hunk 2]
```diff
@@ -36,8 +42,15 @@ TopBar::TopBar(std::shared_ptr<Model::IStream> stream, std::shared_ptr<Model::Me
             [=](const QMediaRecorder::State& state) { onRecorderStateChanged(state); });
     connect(m_ui->recordButton, &QAbstractButton::clicked, [=] { onRecordButtonClicked(); });
 
-    connect(m_ui->meetButton, &QAbstractButton::clicked,
-            [] { QDesktopServices::openUrl(QUrl("https://rendezvous-meet.com/")); });
+    connect(m_ui->meetButton, &QAbstractButton::clicked, [=] { QDesktopServices::openUrl(m_rendezvousMeetUrl); });
+
+    connect(m_transcription.get(), &Model::Transcription::finished, [=](QNetworkReply* reply) {
+        qDebug() << reply->readAll();
```
Comment (Rouiller) [inline steno/src/view/components/top_bar.cpp]: Maybe put this code in a function

File: steno/src/view/components/top_bar.cpp [hunk 3]
```diff
@@ -1,20 +1,26 @@
...
-TopBar::TopBar(std::shared_ptr<Model::IStream> stream, std::shared_ptr<Model::Media> media, QWidget* parent)
+TopBar::TopBar(std::shared_ptr<Model::IStream> stream, std::shared_ptr<Model::Media> media,
+               std::shared_ptr<Model::Transcription> transcription, std::shared_ptr<Model::Config> config,
+               QWidget* parent)
     : QWidget(parent)
     , m_ui(new Ui::TopBar)
     , m_stream(stream)
     , m_media(media)
+    , m_transcription(transcription)
+    , m_config(config)
```
Comment (Rouiller) [inline steno/src/view/components/top_bar.cpp]: Maybe we could store only the transcription config by calling `m_config->transcriptionConfig()` here

File: steno/src/view/components/top_bar.cpp (hunk 1 above)
Comment (alexmo16) [reply steno/src/view/components/top_bar.cpp]: I already finished the real implementation for the next pull request.

File: steno/src/view/components/top_bar.cpp (hunk 2 above)
Comment (alexmo16) [reply steno/src/view/components/top_bar.cpp]: this is different in the next PR

File: steno/src/view/components/top_bar.cpp (hunk 3 above)
Comment (alexmo16) [reply steno/src/view/components/top_bar.cpp]: done

**Pull request merged**